src/utils/artifact/findMissingManifestFiles.ts

```diff
@@ -17,30 +17,32 @@ interface MissingFile {
  */
 export function findMissingManifestFiles(
   manifest: AragonManifest,
-  distPath: string
+  distPath: string,
+  hasFrontend: boolean
 ): MissingFile[] {
   const missingFiles: MissingFile[] = []
 
   function assertFile(filepath: string, id: string, required: boolean): void {
-    // filepath maybe a remote URL, ignore those cases
-    if (filepath.includes('://')) return
-    const fullPath = path.join(distPath, filepath)
-    if (!fs.existsSync(fullPath))
-      missingFiles.push({ path: fullPath, id, required })
+    if (filepath && filepath.includes('://')) {
+      // filepath maybe a remote URL, ignore those cases
+      const fullPath = path.join(distPath, filepath)
+      if (!fs.existsSync(fullPath))
+        missingFiles.push({ path: fullPath, id, required })
+    }
   }
 
   // Assert optional metadata
-  if (manifest.details_url) assertFile(manifest.details_url, 'details', false)
-  manifest.icons.forEach((icon, i) => {
-    assertFile(icon.src, `icon ${i}`, false)
-  })
-  manifest.screenshots.forEach((screenshot, i) => {
-    assertFile(screenshot.src, `screenshot ${i}`, false)
-  })
-
-  // Assert mandatory files
-  assertFile(manifest.start_url, 'start page', true)
-  assertFile(manifest.script, 'script', true)
+  assertFile(manifest.details_url, 'details', false)
+  manifest.icons &&
+    manifest.icons.forEach((icon, i) => {
+      assertFile(icon.src, `icon ${i}`, false)
+    })
+  manifest.screenshots &&
```

Comment:
Let's not abuse the `&&` operator here for an `if` (it is common for rendering react components, for stylistic reasons, but nowhere else because there are more clear alternatives when you don't need to wrap logic in an iife).

Reply:
Cool, thanks for point that out.